pkg/exec/exec.go

```diff
@@ -0,0 +1,73 @@
+package exec
+
+import (
+	"fmt"
+	osexec "os/exec"
+	"strings"
+)
+
+var testMode bool
+var testPrograms map[string]string
+
+type TestResult struct {
+	command string
+	output  string
+	err     error
+}
+
+var testResults []TestResult
+
+func SetTestMode() {
+	testMode = true
+	testPrograms = make(map[string]string)
+}
+
+func AddTestProgram(path string) {
+	lastSlash := strings.LastIndex(path, "/")
+	basename := path[lastSlash+1:]
+	testPrograms[basename] = path
+}
+
+func AddTestResult(command string, output string, err error) {
+	testResults = append(testResults, TestResult{command, output, err})
+}
+
```

Comment:
Looks like except LookPath() and Exec(), rest of the methods are only for testing. Can we move all these test related methods in the end with *Test only methods* header/comment.